src/factories/transactions.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <factories/transactions.h>
+
+CMutableTransaction BuildSpendingTransaction(const CScript& scriptSig, int nValue, const uint256& prevout_hash)
+{
+    CMutableTransaction txSpend;
+    txSpend.nVersion = 1;
+    txSpend.nLockTime = 0;
+    txSpend.vin.resize(1);
+    txSpend.vout.resize(1);
+    txSpend.vin[0].prevout.hash = prevout_hash;
+    txSpend.vin[0].prevout.n = 0;
+    txSpend.vin[0].scriptSig = scriptSig;
+    txSpend.vin[0].nSequence = CTxIn::SEQUENCE_FINAL;
+    txSpend.vout[0].scriptPubKey = CScript();
+    txSpend.vout[0].nValue = nValue;
+
+    return txSpend;
+}
+
+CMutableTransaction BuildSpendingTransaction(const CScript& scriptSig, const CScriptWitness& scriptWitness, int nValue, const uint256& prevout_hash)
+{
+    CMutableTransaction txSpend{BuildSpendingTransaction(scriptSig, nValue, prevout_hash)};
+    txSpend.vin[0].scriptWitness = scriptWitness;
+
+    return txSpend;
+}
+
+CMutableTransaction BuildSpendingTransaction(const CScript& scriptSig, const CMutableTransaction& txCredit)
+{
+    return BuildSpendingTransaction(scriptSig, txCredit.vout[0].nValue, txCredit.GetHash());
```

Comment:
Make the implicit conversion losing conversion here from `long` (`CAmount`) to `int` explicit?